lib/plugin_base.rb

```diff
@@ -43,4 +47,69 @@ def get_markdown_data(markdown_file)
   def format_identifier(identifier)
     identifier.gsub(/((_|-)(ios|android))/i, "")
   end
+
+  private 
+
+  def zapp_plugin
+    plugin_candidates = get_request(plugins_url, request_params)
+      .body
+      .select do |p|
+        p["name"] == @name || identifier_matches?(p)
+      end
+  
+    case plugin_candidates.count
+    when 0
+      color "No Plugin found matching #{@manifest["identifier"]}. please check the identifier and try again", :red
+      color "If you want to create a plugin with a new identifier, use the --new option (see zappifest publish --help)", :red
+      exit
+    when 1
+      plugin_candidates.first
+    else
+      plugin_identifiers = plugin_candidates.map { |p| p["external_identifier"] }
+      identifier_index = multiple_option_question("Please select your plugin", plugin_identifiers)
+      plugin_candidates[identifier_index]
+    end
+  end
+
+  def request_params
+    {}.tap do |params|
+      params["id"] = @existing_plugin["id"] unless @existing_plugin.nil?
+      params["access_token"] = @access_token
+      params["plugin[name]"] = @name
+      params["plugin[category]"] = @manifest["type"]
+      params["plugin[external_identifier]"] = @identifier
+      params["plugin[whitelisted_account_ids][]"] = @manifest["whitelisted_account_ids"] || @plugin_account
```

Comment:
@plugin_account is  a single string - I think it should be converted to an array, something like  `[@plugin_account]`